src/RawHtmlFieldInput.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import {changeBlockValue} from './actions';
+import {
+  getFieldName,
+  isStatic,
+  shouldRunInnerScripts
+} from './processing/utils';
+
+
+const MutationObserver = window.MutationObserver
+                      || window.WebKitMutationObserver
+                      || window.MozMutationObserver;
+
+
+@connect(null, (dispatch, props) => {
+  const {fieldId, blockId} = props;
+  return bindActionCreators({
+    changeBlockValue: value => changeBlockValue(fieldId, blockId, value),
+  }, dispatch);
+})
+class RawHtmlFieldInput extends React.Component {
+  static propTypes = {
+    fieldId: PropTypes.string.isRequired,
+    blockDefinition: PropTypes.object.isRequired,
+    blockId: PropTypes.string.isRequired,
+    html: PropTypes.string.isRequired,
+    value: PropTypes.any,
+  };
+
+  runInnerScripts() {
+    if (shouldRunInnerScripts(this.props.blockDefinition)) {
+      for (let script
+           of ReactDOM.findDOMNode(this).querySelectorAll('script')) {
+        script.parentNode.removeChild(script);
+        window.eval(script.innerHTML);
```

Comment:
I think it would be better to read the scripts and re-inject them, instead of assuming they all contain JS and then executing it here without the original tag. See https://stackoverflow.com/a/47614491/1798491.

Reply:
Also – this should have a `try / catch`, or `componentDidCatch`, and a basic error handling UI. We don't want issues in those scripts to crash the whole StreamField.

Reply:
I tried to inject the scripts in the React HTML using simple `dangerouslySetInnerHTML`. But it does not work. React prevents the scripts from loading. So this `eval` was absolutely the only way I could find to run the JavaScript from Wagtail templates templates. This was one of the things that took me days to figure out… Like you and any good programmer, I hate `eval` ;) But it’s roughly the same as a `<script>[JS]</script>` in terms of security here.

I agree though that I wrongly assumed that they all contained JS. Same about the try/catch.

Reply:
See this StackOverflow discussion for more detail. No-one gets around this issue with `eval`: https://stackoverflow.com/questions/35614809/react-script-tag-not-working-when-inserted-using-dangerouslysetinnerhtml